File: ucloud_joys.js
// Node.js version
var lineReader = require('readline').createInterface({
    input: require('fs').createReadStream('UCloud.txt')
});
var total = 0;
var i = 0;
lineReader.on('line', function (text) {
    var count = (text.match(/UCanUup/g) || []).length;
    console.log('Line', i, ':', count);
    total += count;
    i++;
}).on('close', function () {
    console.log('Total:', total);
});

File: UCloudJoys.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class UCloudJoys {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new FileReader("UCloud.txt"));
		int total = 0;
		int i = 0;
		String text = null;
		while ((text = br.readLine()) != null) {
			int count = text.split("UCanUup").length - 1;
			System.out.println("Line " + i + " : " + count);
			total += count;
			i++;
		}
		System.out.println("Total: " + total);
		br.close();
	}
}